src/sys/boutexception.cxx

```diff
@@ -101,7 +101,7 @@ void BoutException::makeBacktrace() {
   {                                                                                      \
     buflen = 0;                                                                          \
     buffer = nullptr;                                                                    \
-    if (s == nullptr) {                                                                  \
+    if ((s) == nullptr) {                                                                \
```

Comment:
what do the extra brackets do here?

Reply:
In general, macro arguments should be surrounded by brackets as otherwise you may get unintended behaviour when you pass expressions, e.g.:

```cpp
#define SQ(a) (a * a)

SQ(3); // 9
SQ(2 + 1); // -> (2 + 1 * 2 + 1) -> 5
```

Another reason to avoid macros where we can!

In this particular case, I'm not sure there's anything valid you could pass as `s` that would break without the brackets